ark/utils/plot_utils.py

```diff
@@ -28,86 +62,44 @@ def plot_overlay(predicted_contour, plotting_tif, alternate_contour=None, path=N
     predicted_contour_mask[predicted_contour_mask > 0] = 255
 
-    if plotting_tif is None:
-        # will just save the contour mask
-        io.imsave(path, predicted_contour_mask)
-    else:
-        # rescale each channel to go from 0 to 255
-        rescaled = np.zeros(plotting_tif.shape, dtype='uint8')
-
-        for idx in range(plotting_tif.shape[2]):
-            if np.max(plotting_tif[:, :, idx]) == 0:
-                # don't need to rescale this channel
-                pass
-            else:
-                percentiles = np.percentile(plotting_tif[:, :, idx][plotting_tif[:, :, idx] > 0],
-                                            [5, 95])
-                rescaled_intensity = rescale_intensity(plotting_tif[:, :, idx],
-                                                       in_range=(percentiles[0], percentiles[1]),
-                                                       out_range='uint8')
-                rescaled[:, :, idx] = rescaled_intensity
-
-        # overlay first contour on all three RGB, to have it show up as white border
-        rescaled[predicted_contour_mask > 0, :] = 255
-
-        # overlay second contour as red outline if present
-        if alternate_contour is not None:
-
-            if predicted_contour.shape != alternate_contour.shape:
-                raise ValueError("predicted_contour and alternate_"
-                                 "contour array dimensions not equal.")
-
-            # define borders of cell in mask
-            alternate_contour_mask = find_boundaries(alternate_contour, connectivity=1,
-                                                     mode='inner').astype(np.uint8)
-            rescaled[alternate_contour_mask > 0, 0] = 255
-            rescaled[alternate_contour_mask > 0, 1:] = 0
-
-        # save as TIF if path supplied, otherwise display on screen
-        if path is not None:
-            io.imsave(path, rescaled)
-        if show:
-            io.imshow(rescaled)
+    # rescale each channel to go from 0 to 255
+    rescaled = np.zeros(plotting_tif.shape, dtype='uint8')
+
+    for idx in range(plotting_tif.shape[2]):
+        if np.max(plotting_tif[:, :, idx]) == 0:
+            # don't need to rescale this channel
+            pass
+        else:
+            percentiles = np.percentile(plotting_tif[:, :, idx][plotting_tif[:, :, idx] > 0],
+                                        [5, 95])
+            rescaled_intensity = rescale_intensity(plotting_tif[:, :, idx],
+                                                   in_range=(percentiles[0], percentiles[1]),
+                                                   out_range='uint8')
+            rescaled[:, :, idx] = rescaled_intensity
+
+    # overlay first contour on all three RGB, to have it show up as white border
+    rescaled[predicted_contour_mask > 0, :] = 255
+
+    # overlay second contour as red outline if present
+    if alternate_contour is not None:
```

Comment:
Can you try plotting this? I'm pretty sure I coded this wrong. It looks like the second overlay is red, but the DNA channel is also getting put in as red (I think)

Reply:
I can double check, but it looks correct: you're setting the first contour as white by setting all 3 respective channel values as 255. The second contour will be red because you're only setting the first channel as 255, but explicitly set the other 2 channels as 0.

Reply:
Right, I'm saying that the plotting_tif is going to assign DNA to be red. So if we have borders and DNA as red, it won't be that useful

Reply:
Oh ok, I misunderstood. I suppose we can simply assign`alternate_contour_mask` to be a different color, perhaps blue?

Reply:
Because we index into rescaled, the first channel of plotting tif gets assigned to the first channel of rescaled, and the channel order goes RGB. I think it makes more sense for the contour overlay to be red and the DNA to be blue/green

Reply:
@ngreenwald oh OK, so if I'm understanding correctly, this happens in `tif_overlay_preprocess` when we assign `plotting_tif` to take up just the first channel of `formatted_tif` if the third dimension of `plotting_tif` only contains 1 channel. If so, I can just assign it to a different channel. A clarification, though: if `plotting_tif` contains 2 channels, it will be assigned to the red and green channels of `formatted_tif`. Should we leave it like this or assign it differently?

Reply:
As long as we're already raising an informative ValueError, and it's not a numpy error, that's fine. 

Single channel input: assigned to blue
Two channel input: first channel assigned to blue, second channel assigned to green
three channel input: first channel assigned to blue, second channel assigned to green, third channel assigned to red

First overlay assigned to white
Second overlay assigned to red